src/with-in-app-notification.hoc.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import hoistNonReactStatic from 'hoist-non-react-statics';
+
+import Context from './context';
+
+function withInAppNotification(WrappedComponent) {
+  class Enhanced extends React.PureComponent {
+    render() {
+      return (
+        <Context.Consumer>
+          {showNotification => (
+            <WrappedComponent
+              {...this.props}
+              showNotification={showNotification}
+            />
+          )}
+        </Context.Consumer>
+      );
+    }
+  }
+
+  // Pass over static props
+  hoistNonReactStatic(Enhanced, WrappedComponent);
```

Comment:
Could you explain this? I haven't seen this before, thanks :)

Reply:
This is require to avoid stripping the `WrappedComponent` from its `static` methods, you can find more details [here](https://github.com/mridgway/hoist-non-react-statics#readme).
It's a common practice when creating new `HOC`s.

Reply:
Learn something new everyday! I guess I don’t use statics that much so haven’t come across this pattern.